src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -3246,6 +3274,8 @@ because xbuild doesn't support framework reference assemblies.
 	<Delete Files="$(_AndroidAapt2VersionFile)" />
 	<Delete Files="$(IntermediateOutputPath)R.txt" />
 	<Delete Files="$(_AndroidMainDexListFile)" />
+	<Delete Files="$(_AndroidBuildIdFile)" />
+	<Delete Files="$(_ResolvedUserAssembliesHashFile)" />
```

Comment:
Ok, I think since we are handling `FileWrites` properly we don't need these.

You could try the `BuildApplicationAndClean` test, and if it passes without them we're good.